fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/DepositAccountAssembler.java

```diff
@@ -323,10 +324,10 @@ public SavingsAccount assembleFrom(final JsonCommand command, final AppUser subm
                     prodTermAndPreClosure);
 
             FixedDepositAccount fdAccount = FixedDepositAccount.createNewApplicationForSubmittal(client, group, product, fieldOfficer,
-                    accountNo, externalId, accountType, submittedOnDate, submittedBy, interestRate, interestCompoundingPeriodType,
-                    interestPostingPeriodType, interestCalculationType, interestCalculationDaysInYearType, minRequiredOpeningBalance,
-                    lockinPeriodFrequency, lockinPeriodFrequencyType, iswithdrawalFeeApplicableForTransfer, charges,
-                    accountTermAndPreClosure, accountChart, withHoldTax);
+                    accountNo, ExternalIdFactory.produce(externalId), accountType, submittedOnDate, submittedBy, interestRate,
```

Comment:
Here we should generate one if nothing was provided: `ExternalIdFactory.create(...)`